app/search_builders/hyrax/dashboard/managed_search_filters.rb

```diff
@@ -11,15 +11,55 @@ def discovery_permissions
 
       # Override to exclude 'public' and 'registered' groups from read access.
       def apply_group_permissions(permission_types, ability = current_ability)
+        search_terms = add_managing_role_search_filter(ability: ability)
         groups = ability.user_groups
-        return [] if groups.empty?
-        permission_types.map do |type|
+        return search_terms if groups.empty?
+        permission_types.each do |type|
           field = solr_field_for(type, 'group')
-          user_groups = type == 'read' ? groups - [::Ability.public_group_name, ::Ability.registered_group_name] : groups
+          delete_groups = [::Ability.public_group_name, ::Ability.registered_group_name]
+          user_groups = type == 'read' ? groups - delete_groups : groups
           next if user_groups.empty?
-          "({!terms f=#{field}}#{user_groups.join(',')})" # parens required to properly OR the clauses together.
+          # parens required to properly OR the clauses together:
+          search_terms << "({!terms f=#{field}}#{user_groups.join(',')})"
         end
+        search_terms
       end
+
+      # Look for a user's managing role and add filters for all admin sets that have permission
+      # templates that include managing roles.
+      #
+      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/PerceivedComplexity
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def add_managing_role_search_filter(ability:, search_terms: [])
+        search_terms ||= []
```

Comment:
Is setting `search_terms` again here needed? I think `search_terms` has a default value of [] from the arguments list a line above.

Reply:
I was covering the situation that nil was deliberately passed in as the value of :search_term. Does the default value block that possibility? I haven't actually tried that, but I'll test it locally.

Reply:
So ruby passes nil when a named parameter with a non-nil default is called:
`[1] pry(main)> def test_named_parm( parm: [] )
  puts parm.class.name
[1] pry(main)> def test_named_parm( parm: [] )
  puts parm.class.name
end
=> :test_named_parm
[2] pry(main)> test_named_parm
Array
=> nil
[3] pry(main)> test_named_parm( parm: nil )
NilClass
=> nil
`